### ISSUE TITLE
How to use MuationalPatterns (or if I can) with Copy number signatures

### ISSUE THREAD

**Issue by ipstone** [UMCUGenetics/MutationalPatterns#76]
Hi @FreekManders and everyone,

Is there a way to use MutationalPatterns with a customized feature matrix, and customized signature definitions?

I would like to perform signature extraction/decomposition for the recently published Copy number signatures (https://www.nature.com/articles/s41586-022-04738-6 ) - I am wondering if there's a way that we can perhaps similar analyses in the MutationalPatterns package, for copy number analysis?

Thanks a lot!

Isaac

**Comment by FreekManders** [comment]
Yes, kinda. We currently don't have any specific functions for copy numbers, so you will have to create your own feature matrix. If you have a feature matrix and download the cosmic copy number signatures, then you could use functions like `fit_to_signatures`, `extract_signatures`, and `cos_sim_matrix` as these just work on matrices and don't depend on the specific features in the matrix. You will have to visualize the copy number profiles yourself, but that can be done pretty easily with ggplot.

**Comment by ipstone** [comment]
Thanks a lot @FreekManders for these points, I will take a look at these functions/directions.

The package https://github.com/AlexandrovLab/SigProfilerExtractor has a nice function to extract features from copy number caller's output, and there are also visualization plots from https://github.com/AlexandrovLab/SigProfilerPlotting . I will generate my feature matrix from there and try to use MutationalPatterns functions to investigate.
